src/main/java/hu/oe/nik/szfmv/automatedcar/AutomatedCar.java

```diff
@@ -39,6 +47,11 @@ private void calculatePositionAndOrientation() {
         this.setPosX(this.getPosX() + (int)movingVector.getX());
         this.setPosY(this.getPosY() + (int)movingVector.getY());
         setRotation(pt.getAutoSzoge());
+
+        roadSensor.setPos(roadSensor.getPosX() + (int) movingVector.getX(),
```

Comment:
Nem lenne szebb, hogyha kapnának egy referenciát az szülő objektumukra (ami itt egy Automated car) és getterek hozzáadnák az x,y-t a szülő getPosX, getPosY függvényeihez? Akkor nem kellene itt beállítgatni az új pozíciókat

Reply:
Jobb ötlet! A setX és setY állítja a szenzorokat is és akkor itt nem kell

Reply:
A setter átírása méginkább bad practice. Ha megnézel bármilyen játék motort akkor egy elemhez kapcsolt gyermekek mindig a szülőhöz képest adják meg a pozíciójukat, nem a világhoz viszonyítva

Reply:
Lehet igazad van, majd refaktorálhatod ha szeretnéd